### PR TITLE
Binance trade fee inclusion in order

### PR DESCRIPTION
**Pull request by CayenneLow**

### CODE REVIEW

File: xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java [hunk 1]
```diff
@@ -49,6 +50,8 @@
 import org.knowm.xchange.service.trade.params.orders.OrderQueryParams;
 import org.knowm.xchange.utils.Assert;
 
+import lombok.Value;
```
Comment (earce) [inline xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java:53]: is this necessary/used?

File: xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java [hunk 2]
```diff
@@ -321,12 +332,39 @@ public OpenOrdersParams createOpenOrdersParams() {
...
-                    null)));
+        Order order = BinanceAdapters.adaptOrder(
+                        super.orderStatus(
+                          orderQueryParamCurrencyPair.getCurrencyPair(),
+                          BinanceAdapters.id(orderQueryParamCurrencyPair.getOrderId()),
+                          null
+                        )
+                      );
+
+        // Calculate fees by collecting trades associated with this order
```
Comment (earce) [inline xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java:343]: should this logic reside inside of adaptOrder?

File: xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java [hunk 3]
```diff
@@ -321,12 +332,39 @@ public OpenOrdersParams createOpenOrdersParams() {
...
+        BinanceTradeHistoryParams tradeHistParams = new BinanceTradeHistoryParams(orderQueryParamCurrencyPair.getCurrencyPair());
+        UserTrades tradeHist = getTradeHistory(tradeHistParams);
```
Comment (walec51) [inline xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java:345]: scanning the entire trade history for each order will not we acceptable for performance reasons

**Comment by walec51** [comment]
if we cannot pass the orders id to getTradeHistory then this feature is not doable in a performant manner

**Pull request closed without merging**